nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/logviewer/backend/LogFilterForInstance.java

```diff
@@ -178,14 +184,60 @@ public File downloadGivenInstanceLogFile(ServiceLocator habitat, Server targetSe
 
                 wrf.copyTo(instanceLogFile);
             } catch (WindowsException ex) {
-                throw new IOException("Unable to download instance log file from DCOM Instance Node");
+                throw new IOException("Unable to download instance log file from DCOM Instance Node", ex);
             }
 
+        } else if (node.getType().equals("DOCKER") || node.getType().equals("TEMP")) {
+            String dockerContainerId = targetServer.getDockerContainerId();
+            String nodeHost = node.getNodeHost();
+            int nodeDockerPort = Integer.valueOf(node.getDockerPort());
+            try {
+                if (dockerContainerId == null && node.getType().equals("TEMP")) {
+                    // FIXME: it would be easies, if the TEMP server had docker container id set!
+                    // temporary docker node, we need to access the original node, cut the added generated name
+                    String originalInstanceName = instanceName.substring(0, instanceName.lastIndexOf('-'));
+                    Server originalInstance = domain.getServerNamed(originalInstanceName);
+                    dockerContainerId = originalInstance.getDockerContainerId();
```

Comment:
What if the originalInstance is null? If that is the case this will throw NullPointerException

Reply:
This whole part is wrong. We need to get the information -- docker host and port -- from the instance/node, not by searching other instance. So we need to fix creation of the running docker instance+node, fill the information and use it here.